.claude/scripts/plugin-definition-refresh.test.sh

```
@@ -0,0 +1,618 @@
+# Two lanes dispatch hourly, so an unbounded backup set grows forever while nothing reads the old
+# copies. Both halves are asserted: a count alone would pass a prune that kept the OLDEST, which is
+# the one outcome that loses the copy an operator would actually want.
+make_fixture
+set_gitlink "$MK_NEW"
+for stamp in 20260101T000000Z 20260102T000000Z 20260103T000000Z; do
+  echo stale > "$PLUGINS/installed_plugins.json.bak-$stamp-plugin-definition-refresh"
+done
+# An unrelated neighbour must survive: the prune matches only this script's own suffix.
+echo other > "$PLUGINS/installed_plugins.json.bak-20260101T000000Z-someone-else"
+PLUGIN_REFRESH_BACKUP_KEEP=2 STUB_MARKETPLACE_TARGET="$MK_NEW" run >/dev/null 2>&1; rc=$?
+kept="$(ls -1 "$PLUGINS"/installed_plugins.json.bak-*-plugin-definition-refresh 2>/dev/null | wc -l | tr -d ' ')"
```

Shellcheck (0.11.0)

[info] Use find instead of ls to better handle non-alphanumeric filenames.

(SC2012)